tools/clang/test/CodeGenSPIRV/shader.debug.type.matrix.hlsl

```diff
@@ -0,0 +1,10 @@
+// RUN: %dxc -T ps_6_0 -E main -fspv-debug=vulkan -fcgl  %s -spirv | FileCheck %s
+
+
+// CHECK: [[float:%[0-9]+]] = OpExtInst %void {{%[0-9]+}} DebugTypeBasic {{%[0-9]+}} %uint_32 %uint_3 %uint_0
+// CHECK: {{%[0-9]+}} = OpExtInst %void {{%[0-9]+}} DebugTypeVector {{%[0-9]+}} %uint_4
+// CHECK: {{%[0-9]+}} = OpExtInst %void {{%[0-9]+}} DebugTypeMatrix {{%[0-9]+}} %uint_3 %uint_1
+
+void main() {
+   float3x4 mat;
```

Comment:
would it be possible to add 2 more tests:
 - double matrices
 - half matrices (with -enable-16bit-types) (you can add another RUN line with a --check-prefix param to FileCheck to split it
 
```
// RUN: %dxc %s --normal-options | FileCheck %s
// RUN: %dxc --some-other-option | FileCheck %s --check-prefix=CHECK-MY-OTHER-OPTION
// CHECK: some check for the basic RUN
// CHECK-MY-OTHER-OPTION: some check for the new RUN line.
```